### PR TITLE
fix: remove linea mainnet feature toggle

### PR DESCRIPTION
**Pull request by VGau**
## Explanation

Removed Linea mainnet feature toggle and all related functions and conditions.

## Pre-merge author checklist

- [X] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by digiwand — DISMISSED** [review]

**Review by DDDDDanica — DISMISSED** [review]

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by NidhiKJha — COMMENTED** [review]
Looks great! Since, we are removing `isLineaMainnetReleased` from the state do we need a migration here?

**Review by digiwand — DISMISSED** [review]

**Review by pedronfigueiredo — DISMISSED** [review]

File: app/scripts/migrations/101.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  const versionedState = cloneDeep(originalVersionedState);
+  versionedState.meta.version = version;
+
+  const state = versionedState.data;
+  versionedState.data = transformState(state);
+  return versionedState;
+}
+
+function transformState(state: MetaMaskState): MetaMaskState {
+  if (state?.PreferencesController?.isLineaMainnetReleased) {
```
Comment (pedronfigueiredo) [inline app/scripts/migrations/101.ts:32]: This condition is falsy if `isLineaMainnetReleased` is false, but ideally we delete the property in any case. Can you use the method `hasProperty` instead?

Comment (pedronfigueiredo) [inline app/scripts/migrations/101.ts:32]: Can you add this before this condition? This pattern is useful for observing any state inconsistencies through our monitoring system:

```js
 if (
    !hasProperty(state, 'PreferencesController') ||
    !isObject(state.PreferencesController)
  ) {
    captureException(
      `Migration ${version}: Invalid PreferencesController state: ${typeof state.PreferencesController}`,
    );

    return state;
  }
  ```

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by DDDDDanica — DISMISSED** [review]

**Comment by DDDDDanica** [comment]
LGTM !

**Review by DDDDDanica — DISMISSED** [review]

**Comment by gauthierpetetin** [comment]
Hi @VGau , is this PR ready to be reviewed?

File: app/scripts/migrations/104.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 104;
+
+/**
+ * Remove network controller `networkId` state.
```
Comment (danjm) [inline app/scripts/migrations/104.ts:13]: I think this description needs to be updated

**Review by danjm — CHANGES_REQUESTED** [review]
Just one minor change needed related to code comments

File: app/scripts/migrations/104.ts (hunk 2 above)
Comment (VGau) [reply app/scripts/migrations/104.ts]: @danjm I updated the description. Thanks 😄

**Comment by dfox-consensys** [comment]
@gauthierpetetin please can we bump this up the priority list for some clean up. Please lmk if any blockers!

**Comment by gauthierpetetin** [comment]
Hi @dfox-consensys , sure, it's now bumped in the priority 2 section

**Review by pedronfigueiredo — APPROVED** [review]

**Review by DDDDDanica — APPROVED** [review]

**Review by danjm — APPROVED** [review]

**Pull request merged**